packages/cache/README.md

```diff
@@ -1,3 +1,249 @@
+* **Extensible & Layered**: Supplies extension points allowing custom caching
+  implementations (e.g. persistent caching via IndexDB). Users are _not_ limited to
+  out of the box caching configurations.
+* **Zero Config**: Works with sensible default settings "out of the box" with
+  minimal user setup.
+* **Independent**: Caching system is not "aware" of GraphQL semantics or the
+  specifics of how data is loaded (e.g. `fetch` layer, websockets, etc).
+* **Ergonomic Invalidation**: Cache invalidation is notoriously difficult. This
+  system attempts to make invalidation as easy as possible. APIs are designed
+  up front to enable the cache to absorb the complexities of invalidation (and
+  _not_ force that complexity onto the user).
+* **Debuggable**: Exposes the ability to debug caching details. Understanding
+  the sources of the _contents_ of the cache (e.g. [entities](#definitions) being merged as a
+  result of multiple queries) or the reason specific entities are being
+  retained due to caching configuration should not require deep understanding
+  of the cache internals.
+* **Consistent**: A specific [cached entities](#definitions) Identical cached entities loaded across multiple queries are
+  updated "live". Extension points exist to trigger rerendering for various
+  frameworks (e.g. Ember, Glimmer, React, etc).
+
+## Features
+
+* All cache layer APIs must be async
+* Can be extended and composed (e.g. an application could implement persistent caching _on top_ of our default caching system without having to re-implement [entity](#definitions) merging)
+* Supports layered caching
+* Supports merging [cached entities](#definitions)
+  - enable/disable per-request
+  - customizable merging strategies
+  - ID field is configurable by app and by query
+* Supports intelligent merging (implementable in application / infra)
+  - retain as much information as is feasible about the sources of data (likely more information available in development than in production, but _some_ information should still be included in production)
+  - in development/test mode: can identify each of the responses that are merged into a given [entity](#definitions) (including as much information about where those requests actually come from)
+  - provides that information to an application level configurable "[entity](#definitions) merging hook"
+  - can be used to aide debugging
+* [Cached entities](#definitions) and queries can be unloaded in a configurable way with resonable defaults
+* Support caching expiration via multiple mechanisms:
+  - time based (e.g. entities are released after a specific amount of time)
+  - least recently used (e.g. hold on to at least X entities)
+  - ....???
+* Expose low-level API to expire cache entries
+* Expose low-level API for manual cache eviction
+* Exposes public API's to access cache contents for debugging
+  - should this be development mode only??
```

Comment:
Development mode and specific users that are configurable (e.g. LIAM,  LIAS, Security) can access cache contents?

Reply:
The idea would be that they can access the `$debug` APIs directly.

The basic thinking is something like this

1. If possible, `$debug` (i.e. all debugging APIs) are lazily loaded (via `await import`) so that it works in dev + test +prod
2. In general, even when lazily loaded, `$debug` should only be available in a debugging context (e.g. devtools inspectdor) and in particular production code should not be using `$debug` since those APIs are very likely to be slow as they're optimized for detail during debugging.
3. However, **sometimes** it's reasonable to access `$debug` programatically -- e.g. if you want to ship some code to a CI test job to help understand a CI failure.


But actually accessing cache contents is going to be possible via the top-level APIs; the question is about the debugging-specific APIs (that will likely be slower, may well retain a lot more memory &c.)

Reply:
Reviewed the debug API in the other PR, resolving this one